src/etl/common/ETLSession.py

```diff
@@ -19,11 +20,20 @@
 class ETLSession:
     """Spark session class."""
 
-    def __init__(self: ETLSession, cfg: DictConfig) -> None:
+    def __init__(
```

Comment:
Look at this one.
https://github.com/opentargets/genetics_etl_python/pull/43
I think is safer to provide an SparkConf object, than a dictionary

Then you can just do:
```
conf = (
    SparkConf()
    .set(
        "spark.jars",
        "/home/ochoa/.cache/pypoetry/virtualenvs/otgenetics-Izsk1Unf-py3.8/lib/python3.8/site-packages/hail/backend/hail-all-spark.jar",
    )
    .set(
        "spark.driver.extraClassPath",
        "/home/ochoa/.cache/pypoetry/virtualenvs/otgenetics-Izsk1Unf-py3.8/lib/python3.8/site-packages/hail/backend/hail-all-spark.jar",
    )
    .set("spark.executor.extraClassPath", "./hail-all-spark.jar")
    .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
    .set("spark.kryo.registrator", "is.hail.kryo.HailKryoRegistrator")
    .set("spark.kryoserializer.buffer", "512m")
)
```
to then do:

```
etl = ETLSession(cfg, conf)
```

Reply:
I'm not sure about that: if we pass SparkConf, it means you have to deal with spark configurations elsewhere in other modules (`from pyspark.conf import SparkConf`, initializing etc.). Your example would look like this:
```python
hail_spec_configs = {
    "spark.jars": "/home/ochoa/.cache/pypoetry/virtualenvs/otgenetics-Izsk1Unf-py3.8/lib/python3.8/site-packages/hail/backend/hail-all-spark.jar",
    "spark.driver.extraClassPath","/home/ochoa/.cache/pypoetry/virtualenvs/otgenetics-Izsk1Unf-py3.8/lib/python3.8/site-packages/hail/backend/hail-all-spark.jar",
    "spark.executor.extraClassPath": "./hail-all-spark.jar",
    "spark.serializer": "org.apache.spark.serializer.KryoSerializer",
    "spark.kryo.registrator": "is.hail.kryo.HailKryoRegistrator",
    "spark.kryoserializer.buffer": "512m"
}

etl = ETLSession(cfg, hail_spec_configs)
```

But I don't have strong feelings for either solutions. Whatever.